snuba/consumers/consumer.py

```diff
@@ -578,24 +589,32 @@ def create(
         # 2. Filter out any messages that do not apply to any storage.
         # 3. Transform the messages using the selected storages.
         # 4. Route the messages to the collector for each storage.
+        collect = CollectStep(
+            self.__build_collector,
+            commit,
+            self.__max_batch_size,
+            self.__max_batch_time,
+        )
+
+        strategy: ProcessingStrategy[MultistorageKafkaPayload]
+
+        if self.__processes is None:
+            strategy = TransformStep(process_message_multistorage, collect)
+        else:
+            assert self.__input_block_size is not None
+            assert self.__output_block_size is not None
+            strategy = ParallelTransformStep(
+                process_message_multistorage,
+                collect,
+                self.__processes,
+                self.__max_batch_size,
+                self.__max_batch_time,
+                self.__input_block_size,
+                self.__output_block_size,
+                self.__metrics,
+            )
```

Comment:
This is identical to what happens here. I suggest we put that logic in the same place somehow (like externalizing it into its function).
https://github.com/getsentry/snuba/blob/master/snuba/consumers/consumer.py#L337-L369

Reply:
It is not identical, there are some small differences. I think that could be true of much of this class, not just this part and it may make sense to put some of this into a parent class. But I'd rather not change the structure right now.